pkg/phlaredb/symdb/block_writer_v3.go

```diff
@@ -120,18 +112,63 @@ func writePartitionV3(w *writerV3, p *PartitionWriter) (err error) {
 	return nil
 }
 
-func writeSymbolsBlock[T any](w *fileWriter, s []T, e *symbolsEncoder[T]) (h SymbolsBlockHeader, err error) {
-	h.Offset = uint64(w.w.offset)
+func writeSymbolsBlock[T any](w *writerOffset, s []T, e *symbolsEncoder[T]) (h SymbolsBlockHeader, err error) {
+	h.Offset = uint64(w.offset)
 	crc := crc32.New(castagnoli)
-	mw := io.MultiWriter(crc, w.w)
+	mw := io.MultiWriter(crc, w)
 	if err = e.encode(mw, s); err != nil {
 		return h, err
 	}
-	h.Size = uint32(w.w.offset) - uint32(h.Offset)
+	h.Size = uint32(w.offset) - uint32(h.Offset)
 	h.CRC = crc.Sum32()
 	h.Length = uint32(len(s))
 	h.BlockSize = uint32(e.blockSize)
 	h.BlockHeaderSize = uint16(e.blockEncoder.headerSize())
 	h.Format = e.blockEncoder.format()
 	return h, nil
 }
+
+func WritePartitionV3(p *PartitionWriter, dst io.Writer) error {
```

Comment:
I think we may omit the version, as we always want v3 here (or the latest version):

```suggestion
func WritePartition(p *PartitionWriter, dst io.Writer) error {
```